### ISSUE TITLE
PHP 7.4 Support

### ISSUE THREAD

**Issue by alexander-schranz** [jackalope/jackalope-fs#27]
Currently php 7.4 is not supported fully because `key()` function is used on an \ArrayObject here:

https://github.com/jackalope/jackalope-fs/blob/405a0470eb7a7982cb40e723e727f4d94685090f/src/Transport/Fs/Filesystem/Storage/NodeReader.php#L158

Array functions do since 7.4 not longer work on ArrayObjects or iterators ([see](https://www.php.net/manual/de/migration74.incompatible.php#migration74.incompatible.spl)):

```bash
php7.3 -r 'var_dump(key(new \ArrayObject(["test" => "123"])));' // returns string(4) "test"
php7.4 -r 'var_dump(key(new \ArrayObject(["test" => "123"])));' // returns null
```

**Comment by dbu** [comment]
twitter brought me this information: https://twitter.com/slamzoe/status/1204767287934423040

imho unless we can be sure that we always deal with ArrayObject and not a plain array, we should do something like `$key = is_array($referrers) ? key($referrers) : $referrers->iterator()->key();` to be robust.

**Comment by alexander-schranz** [comment]
For documentation the $referrers are always from type NodeCollection as the function called above to get the referrers does new NodeCollection so its not possible to have arrays here.

https://github.com/jackalope/jackalope-fs/blob/405a0470eb7a7982cb40e723e727f4d94685090f/src/Transport/Fs/Filesystem/Storage/NodeReader.php#L157-L158

https://github.com/jackalope/jackalope-fs/blob/acd36ab7b8cf4ee8e5ddca4bb9ab55eb4c9209f0/src/Transport/Fs/Filesystem/Storage/NodeReader.php#L83-L103

**Comment by dbu** [comment]
seems save enough. are you fixing this in #26 or do you want to move it to a separate pull request?

**Comment by dbu** [comment]
fixed in  #26
